ui-spacetimechart/src/utils/snapping.ts

```diff
@@ -0,0 +1,58 @@
+import { type HoveredItem, type Point } from '../lib/types';
+
+/**
+ * This function takes a point and a segment, and returns the point of the segment that is the
+ * closest to the input point.
+ */
+export function getClosestPointOnSegment(
```

Comment:
I see there is no tolerance setting in your snap function, but you are using disks to perform the snap. 
But in some cases the cursor will move from a point on the curve to the end of the segment. And we cannot therefore visualize the data of the curves between this point and the end of the segment.

Reply:
the context:
[Screencast from 2024-08-29 15-47-10.webm](https://github.com/user-attachments/assets/5e13f7a0-ba55-4591-9623-090847a134b7)

Reply:
[Screencast from 2024-08-29 15-47-10.webm](https://github.com/user-attachments/assets/c7cee236-85f6-4b7e-95cf-fbf224a0c859)

Reply:
I get your point. Indeed, when the mouse is too close to a point, the mouse will never snap to the curve. I'm not sure it's an issue, though, since points might be more important than curves (in which case, they must be easier to catch).

What do you think about this issue @thibautsailly?

Reply:
The maximum zoom level displays 1 minute over 96px.
The snapping engages when the cursor is at 12~13px from a point, or 12,5 seconds at this zoom level.
I don't think missing contextual informations for this amount of time could be a problem, but I'll ask engineers on Monday if it's acceptable or not from their point of view.

I'm anticipating that they will be interested snapping at path intersections as well as points.